### PR TITLE
:pencil: Some initial READMEs

### PR DESCRIPTION
**Pull request by janiceblundell**
Initial PR to address [Issue#32](https://github.com/Shopify/web-foundations-private/issues/32).

### CODE REVIEW

File: packages/tasks/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# `@sewing-kit/tasks`
+
+On both a Workspace and a Project, one can execute a set of Tasks. Example tasks are build, dev and test, which apply to both, but lint and type-check tasks can also be run on a Workspace.
+
+A Task is defined with a set of [hooks](/packages/hooks/README.md) and options.
```
Comment (ismail-syed) [inline packages/tasks/README.md:5]: I think we should highlight this first. Something like: 
```
`@sewing-kit/tasks` provides a number of interfaces that define the structure of tasks sewing-kit has knowledge on. This structure is heavily based on interfaces provided by `@sewing-kit/hooks`
```

Comment (ismail-syed) [inline packages/tasks/README.md]: I think we should highlight this first. Something like: 
```
`@sewing-kit/tasks` provides a number of interfaces that define the structure of tasks sewing-kit has knowledge on. This structure is heavily based on interfaces provided by `@sewing-kit/hooks`
```

File: packages/tasks/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# `@sewing-kit/tasks`
+
+On both a Workspace and a Project, one can execute a set of Tasks. Example tasks are build, dev and test, which apply to both, but lint and type-check tasks can also be run on a Workspace.
```
Comment (ismail-syed) [inline packages/tasks/README.md]: I think we need to provide a bit more clarity into the terms here. Perhaps even link out to the interfaces for `WorkspaceTasks` and `ProjectTasks`

```diff
- On both a Workspace and a Project, one can execute a set of Tasks.
+ At a high level, `sewing-kit` represents the tasks applied against a given codebase through `WorkspaceTasks` and `ProjectTasks`. These tasks in-turn, can include a number of other tasks that can represent the tasks associated with building, linting, testing a codebase and more. 
```

File: packages/tasks/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+A Task is defined with a set of [hooks](/packages/hooks/README.md) and options.  
+
+## Installation
+
+```
+yarn add @sewing-kit/tasks --dev
+```
+
+## Usage
```
Comment (ismail-syed) [inline packages/tasks/README.md]: If we don't have any usage examples let's just leave this out.

File: packages/hooks/README.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# `@sewing-kit/hooks`
+
+Hooks are a pluggable way of extending or configuring components of a Workspace (Packages, WebApps, or Services) that we want to perform tasks on.  Core hooks are part of sewing-kit and along with custom hooks they comprise the configuration that defines how these tasks are performed.
```
Comment (ismail-syed) [inline packages/hooks/README.md]: There's an extra space in `.  Core.`

File: packages/hooks/README.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+# `@sewing-kit/hooks`
```
Comment (ismail-syed) [inline packages/hooks/README.md:1]: There's some good documentation on sewing-kit-next's design philosophy on plugins [here](https://github.com/ismail-syed/sewing-kit/blob/main/documentation/plugins.md#hooks) that I think we can leverage.

File: packages/hooks/README.md (hunk 4 above)
Comment (ismail-syed) [inline packages/hooks/README.md]: I think we should emphasize that _hooks_ are what plugins leverage to define functionality exposed to a plugin. A good example of this is [`plugin-eslint`](https://github.com/ismail-syed/sewing-kit/blob/main/packages/plugin-eslint/src/plugin-eslint.ts#L40-L43).

File: packages/hooks/README.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+Both Workspace and Plugin context provide access to the Plugin API that allows you to create a step for a task
+
+## Installation
+
+```
+yarn add @sewing-kit/hooks --dev
+```
+
+## Usage
```
Comment (ismail-syed) [inline packages/hooks/README.md:19]: I think we should dive into some of the capabilities enabled by `SeriesHook` and `WaterfallHook` hook classes with examples of `.hook` and `.run` being used.

**Review by ismail-syed — COMMENTED** [review]
I'm about halfway through this review. I need to step out for a bit. This is a great start we have. As you transition into your new team I don't expect all these comments to get addressed. My comments were more posterity.

File: packages/hooks/README.md (hunk 6 above)
Comment (janiceblundell) [reply packages/hooks/README.md:19]: Since your review, I added to the above.  Will also find some examples.

**Comment by b-ryu** [comment]
Core SK-next pages:
- [x] `cli`
- [x] `tasks`
- [ ] `core`
- [ ] `plugins`
- [ ] `hooks`
- [ ] `model`

**Comment by ismail-syed** [comment]
Closing in favour of https://github.com/Shopify/sewing-kit-next/pull/31

**Pull request closed without merging**